src/client/editor/MeshCombinationGroup.js

```diff
@@ -0,0 +1,165 @@
+import THREE from "./three";
+import { isStatic, computeAndSetStaticModes } from "./StaticMode";
+
+function getBase64Image(img) {
+  // Create an empty canvas element
+  const canvas = document.createElement("canvas");
+  canvas.width = img.width;
+  canvas.height = img.height;
+
+  // Copy the image contents to the canvas
+  const ctx = canvas.getContext("2d");
+  ctx.drawImage(img, 0, 0);
+
+  // Get the data-URL formatted image
+  // Firefox supports PNG and JPEG. You could check img.src to
+  // guess the original format, but be aware the using "image/jpg"
+  // will re-encode the image.
+  const dataURL = canvas.toDataURL("image/png");
+
+  return dataURL.replace(/^data:image\/(png|jpg);base64,/, "");
+}
+
+export function compareImages(a, b) {
+  if (a && b) {
+    return a.src === b.src || getBase64Image(a) === getBase64Image(b);
```

Comment:
actually caching these may actually be something to be cautious about -- it would (I think) cause a large relative increase in memory pressure vs the way it is now

Reply:
you could cache the sha of the base64 image instead